### PR TITLE
Move pane collapse state to redux and back up in prefs system

### PR DESCRIPTION
**Pull request by wldcordeiro**
Associated Issue: #1587 

### Summary of Changes

* refactor pane collapse state to use reducers
* back up the state in `prefs`

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Open debugger
- [x] collapse a pane
- [x] refresh and see that pane collapse state is retained

### CODE REVIEW

File: src/reducers/ui.js [hunk 1]
```diff
@@ -30,6 +35,13 @@ function update(state = State(), action: Action): Record<UIState> {
     case constants.SHOW_SOURCE: {
       return state.set("shownSource", action.sourceUrl);
     }
+
+    case constants.TOGGLE_PANE: {
+      prefs[`${action.position}PanelCollapsed`] = action.paneCollapsed;
```
Comment (jasonLaster) [inline src/reducers/ui.js]: nit: could be one `panelCollapsed = \`${action.position}PanelCollapsed\`` var

**Comment by jasonLaster** [comment]
Hey @wldcordeiro I pushed some changes to make the reducer less magical with concatenation because flow got scared. I think if you finish this up flow will be happy.

**Comment by wldcordeiro** [comment]
@jasonLaster Flow is being weird, the issues left were a line that was too long and a type that was used but didn't exist. Oddly enough though I still see two errors from Flow but one is from something already in the codebase and the other is the same error but for some new code.

**Comment by jasonLaster** [comment]
@wldcordeiro lets look at it tomorrow...

**Comment by wldcordeiro** [comment]
@jasonLaster cool, that's fine. I'm still not sure what Flow is complaining about, there's nothing special about the types from what they were before. 🙁

File: src/reducers/ui.js [hunk 2]
```diff
@@ -37,9 +37,13 @@ function update(state = State(), action: Action): Record<UIState> {
...
-      const paneCollapsed = `${action.position}PanelCollapsed`;
-      prefs[paneCollapsed] = action.paneCollapsed;
-      return state.set(paneCollapsed, action.paneCollapsed);
+      if (action.position == "start") {
+        prefs.startPanelCollapsed = action.paneCollapsed;
+        return state.set("startPanelCollapsed", action.paneCollapsed);
+      }
+
+      prefs.endPanelCollapsed = action.paneCollapsed;
+      return state.set("startPanelCollapsed", action.paneCollapsed);
```
Comment (wldcordeiro) [inline src/reducers/ui.js]: `start` -> `end` here

**Comment by jasonLaster** [comment]
💃 

@bomsy, @wldcordeiro we finally got the pref patches in. 

Wellington, the problem was that we weren't adding the default-prefs to the new file

**Pull request merged**